### PR TITLE
Introduce BigqueryJob, like BigqueryTable etc

### PR DESCRIPTION
**Pull request by jskeet**
Additionally, allow jobs to be listed from outside the client's project.

### CODE REVIEW

File: src/Google.Bigquery.V2/BigqueryJob.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        /// and job ID components.
+        /// </remarks>
+        public JobReference Reference => _apiResource.JobReference;
+
+        // TODO: Use an enum instead? Definitely nicer than magic strings.
+
+        /// <summary>
+        /// The state of the job: "RUNNING", "PENDING" or "DONE".
+        /// </summary>
+        public string State => _apiResource.Status.State;
```
Comment (chrisdunelm) [inline src/Google.Bigquery.V2/BigqueryJob.cs]: Enum'erize this state?

**Comment by chrisdunelm** [comment]
Generally LGTM

File: src/Google.Bigquery.V2/BigqueryJob.cs (hunk 1 above)
Comment (jskeet) [inline src/Google.Bigquery.V2/BigqueryJob.cs]: Unsure at the moment. Maybe - I definitely want to revisit the whole aspect of how and what we expose from the underlying resource. I've got a TODO a couple of lines earlier already :)

**Comment by jskeet** [comment]
Thanks - will merge now, and we can think more about the state (etc).

**Pull request merged**